backend/otone_client.py

```diff
@@ -5,59 +5,95 @@
 
+import asyncio
+import logging
+import logging.handlers
+import os
+import shutil
+import sys
+import time
 
-"""
+from autobahn.wamp.serializer import JsonSerializer, MsgPackSerializer
 
+# If code is frozen (i.e. pyinstaller executable) then
+# file path is the sys._MEIPASS attribute
+if getattr(sys, 'frozen', None):
+    print('using frozen path...')
+    path = dir_path = sys._MEIPASS
+    perm_dir_path = '~/Opentrons-Config'
+else:
+    print('using non frozen path...')
+    path = os.path.abspath(__file__)
+    dir_path = os.path.dirname(path)
+    perm_dir_path = dir_path
 
-from autobahn.wamp.serializer import JsonSerializer, MsgPackSerializer
+if len(sys.argv) > 1:
+    perm_dir_path = sys.argv[1]
 
-#import RobotLib
-import json, asyncio, sys, time, collections, os, sys, shutil
+disconnect_counter = 0
+disconnect_count_threshold = 12
+keep_backend_running = False
 
-import logging
+if len(sys.argv) > 2:
+    keep_backend_running = True
+
+print('resource path is', path)
+print('dir path is', dir_path)
+print('perm path is', perm_dir_path)
+
+if not os.path.exists(perm_dir_path):
+    os.makedirs(perm_dir_path)
 
-#for testing purposes, read in a protocol.json file
-path = os.path.abspath(__file__)
-dir_path = os.path.dirname(path)
-dir_par_path = os.path.dirname(dir_path)
-dir_par_par_path = os.path.dirname(dir_par_path)
 fname_default_protocol = os.path.join(dir_path,'data/sample_user_protocol.json')
 fname_default_containers = os.path.join(dir_path, 'data/containers.json')
 fname_default_calibrations = os.path.join(dir_path, 'data/pipette_calibrations.json')
-fname_data = os.path.join(dir_path,'otone_data')
-fname_data_logfile = os.path.join(dir_path,'otone_data/logfile.txt')
-fname_data_containers = os.path.join(dir_path,'otone_data/containers.json')
-fname_data_calibrations = os.path.join(dir_path, 'otone_data/pipette_calibrations.json')
+
+fname_data = os.path.join(perm_dir_path,'otone_data')
+
+fname_data_logfile = os.path.join(perm_dir_path,'otone_data/logfile.txt')
+fname_data_containers = os.path.join(perm_dir_path,'otone_data/containers.json')
+fname_data_calibrations = os.path.join(perm_dir_path, 'otone_data/pipette_calibrations.json')
 
 
 if not os.path.isdir(fname_data):
     os.makedirs(fname_data)
-#if not os.path.exists(fname_data_containers):
-open(fname_data_containers,"w+")
+
+open(fname_data_containers, "w+")
 shutil.copy(fname_default_containers, fname_data_containers)
 
 if not os.path.exists(fname_data_calibrations):
-    open(fname_data_calibrations,"w+")
+    open(fname_data_calibrations, "w+")
     shutil.copy(fname_default_calibrations, fname_data_calibrations)
 
-FORMAT = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
-logging.basicConfig(filename='logfile.txt', level=logging.DEBUG, format=FORMAT)
-logging.info('\n\nOT.One Started')
+
+print('logging to {}'.format(fname_data_logfile))
+
+logger = logging.getLogger('app')
+
+handler = logging.handlers.RotatingFileHandler(
```

Comment:
For this stretch of code; i'm thinking we can create a function in a utils module that gets you a logger and adds all the handlers, formatters, etc.. but we can come back to this later... In general just trying think about the prep of the logger so other modules don't have to config it every time..